File: practica4/CODIGO/C++/Laplacina/laplaciana.cpp
#include <iostream>
#include <opencv2/imgproc.hpp>
#include <opencv2/highgui.hpp>

using namespace std;
using namespace cv;

int main(){
string filename = "../../../data/images/sample.jpg";

// Lemos a imaxe
Mat image = imread(filename, IMREAD_GRAYSCALE);

Mat laplacian, LOG;
int kernelSize = 3;

// Aplicamos a laplacian
Laplacian(image, laplacian, CV_32F, kernelSize, 1, 0);

Mat img1;

GaussianBlur(image, img1, Size(3,3), 0, 0);

// Normalizamos
normalize(laplacian, laplacian, 0, 1, NORM_MINMAX, CV_32F);

imshow("Laplaciana",laplacian);
waitKey(0);

return 0;
}

File: codigos_clase/practica4/CODIGO/PYTHON/Laplaciana/laplaciana.py
import cv2
import numpy as np

img = cv2.imread("../../data/images/sample.jpg", cv2.IMREAD_GRAYSCALE)
# Comprobamos se se leu a imaxe
if img is None:
    print("Non poiden ler a imaxe")
    
kernelSize = 3

# Aplicamos a Laplaciana
img1 = cv2.GaussianBlur(img,(3,3),0,0)
laplacian = cv2.Laplacian(img1, cv2.CV_32F, ksize = kernelSize,
                            scale = 1, delta = 0)

# Normalizamos os resultados
cv2.normalize(laplacian,
                dst = laplacian,
                alpha = 0,
                beta = 1,
                norm_type = cv2.NORM_MINMAX,
                dtype = cv2.CV_32F)

cv2.imshow("Laplaciana", laplacian)
cv2.waitKey(0)
